enclave-runner/src/usercalls/mod.rs

```diff
@@ -144,19 +144,26 @@ where
     }
 }
 
-trait SyncStream: 'static + Send + Sync {
-    fn read_alloc(&self, out: &mut OutputBuffer) -> IoResult<()> {
-        let mut buf = [0u8; 8192];
+/// This trait is mostly same as [Read](https://doc.rust-lang.org/nightly/std/io/trait.Read.html) + [Write](https://doc.rust-lang.org/nightly/std/io/trait.Write.html) except that it takes an immutable reference to the source.
```

Comment:
It's hard to have this automatically link to the correct version of the std docs. I'd suggest not including the links, e.g.:
```
... same as `std::io::Read + std::io::Write` except ...
```